Marlin/src/lcd/extui/lib/anycubic_chiron/chiron_tft_defs.h

```diff
@@ -87,19 +87,19 @@
 #define  AC_msg_version                PSTR("J27")
 
 #define MARLIN_msg_start_probing       PSTR("Probing Point 1/25")
-#define MARLIN_msg_probing_failed      PSTR("Probing Failed")
-#define MARLIN_msg_ready               PSTR(" Ready.")
-#define MARLIN_msg_print_paused        PSTR("Print Paused")
-#define MARLIN_msg_print_aborted       PSTR("Print Aborted")
-#define MARLIN_msg_extruder_heating    PSTR("E Heating...")
-#define MARLIN_msg_bed_heating         PSTR("Bed Heating...")
-
-#define MARLIN_msg_nozzle_parked       PSTR("Nozzle Parked")
-#define MARLIN_msg_heater_timeout      PSTR("Heater Timeout")
-#define MARLIN_msg_reheating           PSTR("Reheating...")
-#define MARLIN_msg_reheat_done         PSTR("Reheat finished.")
-#define MARLIN_msg_filament_purging    PSTR("Filament Purging...")
-#define MARLIN_msg_special_pause       PSTR("PB")
+#define MARLIN_msg_probing_failed      "Probing Failed"_hash
```

Comment:
So, again, here we see a clear shortcoming of the strategy of comparing the status message to a string (or these hashes) in order to figure out the status of the printer. The status messages are translatable strings, so they will not always be in English. Therefore the first thing to do is to get rid of this technique that the Chiron TFT is using and devise a better way for it to get the status updates it's looking for. Other displays manage to switch screens or update the text without a hawkish hook into the changing of the status message. If ExtUI doesn't currently contain the "`onSomeEvent`" handler needed for the events Chiron cares about, then ExtUI can be extended.

Reply:
You're right and you know a lot more than me here. Can we make this change ? Those strings seems like they are only used here, so I don't really know where they come from.

Reply:
Is there a event broadcaster system in Marlin, it would simplify a lot of these codes (found in 2 UI it seems) ?

Reply:
The `ExtUI` API works on a callback system, but it might not yet have all the hooks the developer was looking for. That's okay, because it is always being extended to fit different needs.

To follow up on the Chiron code I would look at each of the status messages that the Chiron is acting upon, find out where that message is being set, and make sure there is a suitable `ExtUI` hook being called at that point. If there is no hook, I would add one. Finally, I would modify the Chiron code to implement those hooks instead of examining the status message.